catalog/plugin/catalog-plugin-querymonitor/src/test/java/ddf/catalog/querymonitor/QueryMonitorTest.java

```diff
@@ -0,0 +1,243 @@
+        as = new ActiveSearch(null, null, null, null);
+        assertThat(qmpi.addActiveSearch(as), is(true));
+    }
+
+    @Test
+    public void testQueryMonitorPluginImplRemoveActiveSearchUsingUUIDNullUUID()
+            throws StopProcessingException, PluginExecutionException {
+        QueryRequest mockQR = mock(QueryRequest.class);
+        propertyMap = new ConcurrentHashMap<>();
+        when(mockQR.getProperties()).thenReturn(propertyMap);
+        when(mockQR.toString()).thenReturn(QUERY_REQUEST_TEXT);
+        qmpi.setRemoveSearchAfterComplete(true);
+        qmpi.process(null, mockQR);
+        Map<UUID, ActiveSearch> activeSearchTable = qmpi.getActiveSearches();
+        if (activeSearchTable != null && !activeSearchTable.isEmpty()) {
+            Iterator it = activeSearchTable.entrySet()
+                    .iterator();
+            if (it != null && it.hasNext()) {
+                Map.Entry<UUID, ActiveSearch> tempEntry = (Map.Entry) it.next();
+                as = tempEntry.getValue();
+                UUID u = null;
+                assertThat(qmpi.removeActiveSearch(u), is(false));
+                u = as.getUniqueID();
+                qmpi.removeActiveSearch(u);
+                return;
+            }
+        }
+        Assert.fail();
+    }
+
+    @Test
+    public void testQueryMonitorPluginImplRemoveActiveSearchUsingUUIDValidUUID()
+            throws StopProcessingException, PluginExecutionException {
+        QueryRequest mockQR = mock(QueryRequest.class);
+        propertyMap = new ConcurrentHashMap<>();
+        when(mockQR.getProperties()).thenReturn(propertyMap);
+        when(mockQR.toString()).thenReturn(QUERY_REQUEST_TEXT);
+        qmpi.setRemoveSearchAfterComplete(true);
+        qmpi.process(null, mockQR);
+        Map<UUID, ActiveSearch> activeSearchTable = qmpi.getActiveSearches();
+        if (activeSearchTable != null && !activeSearchTable.isEmpty()) {
+            Iterator it = activeSearchTable.entrySet()
+                    .iterator();
+            if (it != null && it.hasNext()) {
+                Map.Entry<UUID, ActiveSearch> tempEntry = (Map.Entry) it.next();
+                as = tempEntry.getValue();
+                UUID u = as.getUniqueID();
+                assertThat(qmpi.removeActiveSearch(u), is(true));
+                return;
+            }
+        }
+        Assert.fail();
+    }
+
+    @Test
+    public void testQueryMonitorAddActiveSearchNullParameters() {
+        QueryRequest mockQR = mock(QueryRequest.class);
+        propertyMap = new ConcurrentHashMap<>();
+        when(mockQR.getProperties()).thenReturn(propertyMap);
+        when(mockQR.toString()).thenReturn(QUERY_REQUEST_TEXT);
+        as = new ActiveSearch(null, null, null, null);
+        qmpi.addActiveSearch(as);
+        Map<UUID, ActiveSearch> activeSearchTable = qmpi.getActiveSearches();
+        assertThat(activeSearchTable.containsValue(as), is(true));
```

Comment:
Hamcrest has matchers for testing map contents: http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/collection/IsMapContaining.html#hasValue(V)